core/types.ml

```diff
@@ -3124,11 +3128,15 @@ module RoundtripPrinter : PRETTY_PRINTER = struct
             method effect_row : row -> 'self_type * row_erasability * tid option
               = let maybe_contract : typ -> typ
                   = function
-                  | Operation (d,c) as tp when ES.contract_operation_arrows policy ->
+                  | Operation (d,c,_) as tp when ES.contract_operation_arrows policy ->
                      (* we are in an operation fieldspec and it's an arrow, and we
                       want contractions: check if a contraction is possible here
                       *)
                      if d = unit_type then c else tp
+                     (* FIXME: WT: I dont understand where this is
```

Comment:
Yes, this should probably be updated. This logic here is used for pretty-printing types. It reintroduces some of the sugar that was elaborated by earlier passes.

Reply:
One possible easy change here is to use `tp` instead of `c` for linear nullary operations. For example, `Hi : Bool` is an unlimited nullary operation, and `Hi : () =@ Bool` is a linear nullary operation.

Reply:
Yes, maybe we should get rid of the nullary sugar altogether... But as a first step lets be explicit when the operation interpretation is linear.